app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesFragment.kt

```diff
@@ -196,6 +203,45 @@ class PreferencesFragment : PreferenceFragmentCompat(), Injectable {
                 }
             }
 
+            preferenceCategory(R.string.pref_title_wellbeing_mode) {
+                switchPreference {
+                    title = getString(R.string.limit_notifications)
+                    setDefaultValue(false)
+                    key = PrefKeys.WELLBEING_LIMITED_NOTIFICATIONS
+                    setOnPreferenceChangeListener { _, value ->
+                        accountManager.activeAccount?.let { account ->
+                            val notificationFilter = deserialize(account.notificationsFilter).toMutableSet()
+
+                            if (value == true) {
+                                notificationFilter.add(Notification.Type.FAVOURITE)
+                                notificationFilter.add(Notification.Type.FOLLOW)
+                                notificationFilter.add(Notification.Type.REBLOG)
+                            } else {
+                                notificationFilter.remove(Notification.Type.FAVOURITE)
+                                notificationFilter.remove(Notification.Type.FOLLOW)
+                                notificationFilter.remove(Notification.Type.REBLOG)
+                            }
+
+                            account.notificationsFilter = serialize(notificationFilter)
+                            accountManager.saveAccount(account)
```

Comment:
This is weird: general settings should affect ALL accounts. This only changes the current account. I'd argue this is not the right place to do it, rather leave the accounts alone and change the filters when they are retrieved based on the wellbeing setting.

Reply:
I didn't wanted to straight up ignoring the user-set filter, but I definitely see your concern. What if we update the settings for all accounts in the app, as opposed to just the active one?

Reply:
that is also a possibility 👍